cli/azd/internal/repository/app_init.go

```diff
@@ -1107,12 +1112,61 @@ func promptSpringBootVersion(console input.Console, ctx context.Context) (string
 	}
 }
 
+func promptMissingEventHubsNameOrExit(console input.Console, ctx context.Context, eventHubs *appdetect.AzureDepEventHubs) {
+	switch eventHubs.FromDependency {
+	case appdetect.SpringIntegrationEventHubs, appdetect.SpringMessagingEventHubs, appdetect.SpringKafka:
+		eventHubsName, err := promptEventHubsName(console, ctx)
+		if err != nil {
+			console.Message(ctx, fmt.Sprintf("Error happened when prompt eventhubs name: %s.", err))
+			os.Exit(-1)
+		}
+		eventHubs.EventHubsNamePropertyMap = map[string]string{
+			string(eventHubs.FromDependency): eventHubsName,
+		}
+	case appdetect.SpringCloudStreamEventHubs, appdetect.SpringCloudStreamKafka:
+		promptMissingPropertyAndExit(console, ctx, "spring.cloud.stream.bindings.<binding name>.destination")
+		os.Exit(0)
+	case appdetect.SpringCloudEventHubsStarter:
+		promptMissingPropertyAndExit(console, ctx, "spring.cloud.azure.eventhubs.event-hub-name")
```

Comment:
Could you please help to confirm these things:
1. Do we need to detect properties like `spring.cloud.azure.eventhubs.processor.event-hub-name`.
2. Will `spring.cloud.azure.eventhubs.event-hub-name` be used as `spring.cloud.azure.eventhubs.processor.event-hub-name` when the later one is not configured.

Reply:
Following the [doc](https://learn.microsoft.com/azure/developer/java/spring-framework/configuration-properties-azure-event-hubs) , we can specify the `event-hub-name` property in different levels,
+ spring.cloud.azure.eventhubs.event-hub-name
+ spring.cloud.azure.eventhubs.producer.event-hub-name
+ spring.cloud.azure.eventhubs.consumer.event-hub-name
+ spring.cloud.azure.eventhubs.processor.event-hub-name

After confirmed with Moary, the sub-level property should override the base level property. So for your comment,

> Do we need to detect properties like `spring.cloud.azure.eventhubs.processor.event-hub-name`.

Yes, I think we need.

> Will `spring.cloud.azure.eventhubs.event-hub-name` be used as `spring.cloud.azure.eventhubs.processor.event-hub-name` when the later one is not configured.

It should be used, but now the behavior is not expected. I have created an issue to request double confirm from Moary.
https://github.com/Azure/azure-sdk-for-java/issues/43593

Reply:
Great investigation! Thanks for the detailed explanation.
Please track these things:

> Yes, I think we need.

Support it in another PR.

> It should be used, but now the behavior is not expected.

Please make SJAD work well no matter the problem is fixed or not in Spring Cloud Azure.

In case you forget these things, I suggest you to create an ADO item.

Reply:
> Support it in another PR.
I have updated this PR to support to detect these properties and extract the event-hub names. In SJAD, we only care about event-hub-name.

> Please make SJAD work well no matter the problem is fixed or not in Spring Cloud Azure.
Yeah, after support detecting these properties, SJAD can work with them. The SDK issue should be another separate one.